cms/bundles/signal_handlers.py

```diff
@@ -39,3 +50,128 @@ def handle_bundle_publication(instance: Bundle, **kwargs: Any) -> None:
+
+    client = BundleAPIClient()
+
+    try:
+        # Add the content item to the bundle
+        content_item = build_content_item_for_dataset(instance.dataset)
+
+        response = client.add_content_to_bundle(instance.parent.bundle_api_id, content_item)
+
+        # Extract content_id from the response
+        content_id = extract_content_id_from_bundle_response(response, instance.dataset)
+
+        if content_id:
+            instance.content_api_id = content_id
+            instance.save(update_fields=["content_api_id"])
+            logger.info(
+                "Added content %s to bundle %s in Dataset API with ID %s",
+                instance.dataset.namespace,
+                instance.parent.pk,
+                content_id,
+            )
+        else:
+            logger.error("Could not find content_id in response for bundle %s", instance.parent.pk)
+
+    except BundleAPIClientError as e:
+        logger.error("Failed to add content to bundle %s in Dataset API: %s", instance.parent.pk, e)
+
+
+@receiver(post_delete, sender=BundleDataset)
+@ons_bundle_api_enabled
+def handle_bundle_dataset_removed(instance: BundleDataset, **kwargs: Any) -> None:
+    """Handle when a dataset is removed from a bundle."""
+    if not instance.parent.bundle_api_id or not instance.content_api_id:
+        return
+
+    client = BundleAPIClient()
+
+    try:
+        content_id = instance.content_api_id
+        client.delete_content_from_bundle(instance.parent.bundle_api_id, content_id)
+        logger.info("Deleted content %s from bundle %s in Dataset API", content_id, instance.parent.pk)
+    except BundleAPIClientError as e:
+        logger.error("Failed to delete content %s from bundle %s in Dataset API: %s", content_id, instance.parent.pk, e)
+        # Don't raise the exception to avoid breaking the deletion process
+
+
+@receiver(post_delete, sender=Bundle)
+@ons_bundle_api_enabled
+def handle_bundle_deletion(instance: Bundle, **kwargs: Any) -> None:
+    """Handle when a bundle is deleted."""
+    if instance.bundle_api_id:
+        client = BundleAPIClient()
+
+        try:
+            client.delete_bundle(instance.bundle_api_id)
+            logger.info("Deleted bundle %s from Dataset API", instance.pk)
+
+        except BundleAPIClientError as e:
+            logger.error("Failed to delete bundle %s from Dataset API: %s", instance.pk, e)
+
+
+def _sync_preview_teams_with_bundle_api(bundle: Bundle) -> None:
+    """Helper function to sync a bundle's preview teams with the Bundle API."""
+    if not bundle.bundle_api_id:
+        # This happens when no datasets have been added to the bundle yet
+        logger.info("Bundle %s does not have an API ID. Skipping preview team sync.", bundle.pk)
+        return
+
+    payload = {"preview_teams": get_preview_teams_for_bundle(bundle)}
+
+    try:
+        client = BundleAPIClient()
+        client.update_bundle(bundle_id=bundle.bundle_api_id, bundle_data=payload)
+        logger.info(
```

Comment:
suggestion: log our bundle ID alongside the bundle API id. It will be easier for the support team to get to our bundle with the Wagtail id